### PR TITLE
 Exemplar how to download other instructLab curated models like granite and use as default

### LINKED ISSUE #1282: Download instructions needed under `ilab download` for non-default, curated models (e.g., granite)
**Is your feature request related to a problem? Please describe.**
The taxonomy meetings have had questions/confusion on how to download/use granite models in place of the default model coded into ilab.  Primarily they reference the granite model(s) (e.g., ) referenced and published by ilab itself (i.e., https://huggingface.co/instructlab)

**Describe the solution you'd like**
A simple example on how to download the the granite model and place it into the `models` directory (by convention) would be a huge help to users who want to start using non-default models with granite being a key one we reference repeatedly.  

This example would provide the pattern (plug-and-play) for other models the project intends to adopt/publish and subsequently be used as the "default" for other commands. That is, the example should include additional help/refs. etc. to show how `ilab init` can (after download) be used to ref. the model as the new default.

**Additional context**
The goal of doc. changes is intended to be limited to the use case described above (i.e., to apply to instructLab curated models and not be exhaustive in describing how to download other models which may require tokens or bring in other complexities or entanglements.

### PR DESCRIPTION
**Pull request by mrutkows**
# Changes

During a recent taxonomy meeting, some attendees wanted to use the `granite` model (non-default, but curated/supported by instructLab) and were confused on how to download it and make it avail. to the other commands they used (in order to test new, prospective Q&A skills).  As this seemed to be a common concern for attendees, I promised to provide a PR to add this info. to the docs. to hopefully help them and others via a docs. PR.

**Which issue is resolved by this Pull Request:**
Resolves #1282 

**Description of your changes:**

Added and explained how one would download another supported model, i.e., "granite", into the `models` subdir and subsequently use `ilab init` to make it the default by reference.

**Note** This is a third attempt at the PR I closed in order to account for merge conflicts and have a clean review/discussion. see:
- https://github.com/instructlab/instructlab/pull/1232
- https://github.com/instructlab/instructlab/pull/1284

My lesson-to-self is to not mix using GitHub UI with command line...

### CODE REVIEW

**Review by leseb — CHANGES_REQUESTED** [review]
The commit message is still empty. Please incorporate the PR description into it. It's just:

```
$ git commit --amend
<add the PR description>
$ git push -f origin dload2
```

**Comment by bjhargrave** [comment]
> The commit message is still empty

I see a commit message which matches the PR title.

**Review by bjhargrave — APPROVED** [review]

**Comment by leseb** [comment]
> > The commit message is still empty
> 
> I see a commit message which matches the PR title.

Look at the body of https://github.com/instructlab/instructlab/pull/1334/commits/099b9b6548ad31bef0134a13fb49f23b6096d407, it's empty and does not reflect the "Changes" section of the PR.

File: README.md [hunk 1]
```diff
@@ -331,17 +331,29 @@ For an overview of the full workflow, see the [workflow diagram](./docs/workflow
 
   > **NOTE** ⏳ This command can take few minutes or immediately depending on your internet connection or model is cached. If you have issues connecting to Hugging Face, refer to the [Hugging Face discussion forum](https://discuss.huggingface.co/) for more details.
 
+  #### Downloading another instructLab model from Hugging Face
```
Comment (nathan-weinberg) [inline README.md:334]:
```suggestion
  #### Downloading another InstructLab model from Hugging Face
```

File: README.md [hunk 2]
```diff
@@ -320,7 +320,7 @@ For an overview of the full workflow, see the [workflow diagram](./docs/workflow
   ilab download
   ```
 
-  `ilab download` downloads a compact pre-trained version of the [model](https://huggingface.co/instructlab/) (~4.4G) from HuggingFace and store it in a `models` directory:
+  By default, `ilab download` downloads a compact pre-trained version of the [model](https://huggingface.co/instructlab/) (~4.4G) from HuggingFace and store it in a `models` directory:
```
Comment (nathan-weinberg) [inline README.md:323]: When we say "the model" we are talking about Merlinite yes? Should we explicitly call that out here?

File: README.md [hunk 3]
```diff
@@ -331,17 +331,29 @@ For an overview of the full workflow, see the [workflow diagram](./docs/workflow
...
 
+  #### Downloading another instructLab model from Hugging Face
+
+  If you wish to download another [InstructLab model](https://huggingface.co/instructlab) from Hugging Face, for example the [`instructlab/granite-7b-lab-GGUF`](https://huggingface.co/instructlab/granite-7b-lab-GGUF) model, you can use this command with additional parameters as follows:
+
+  ```shell
+  ilab download --repository instructlab/granite-7b-lab-GGUF --filename granite-7b-lab-Q4_K_M.gguf --model-dir models
+  ```
+
+  This will place the downloaded model into the project's `./models` directory from which you can reference it on other commands.  To make this newly downloaded model the default, simply rerun the [`ilab init`](#️-initialize-ilab) command and when prompted enter `models/granite-7b-lab-Q4_K_M.gguf` for the model path.
```
Comment (nathan-weinberg) [inline README.md:342]:
```suggestion
  This will place the downloaded model into the `models` directory from which you can reference it on other commands.  To make this newly downloaded model the default, simply rerun the [`ilab init`](#️-initialize-ilab) command and when prompted enter `models/granite-7b-lab-Q4_K_M.gguf` for the model path.
```

**Review by nathan-weinberg — CHANGES_REQUESTED** [review]
Thanks for this addition @mrutkows!

**Comment by drpaneas** [comment]
Please let me know if the following is a bug or wrong usage from my part:

0. Get a fresh InstructLab environment

1. Initialize it using the granite model

```
ilab init --model-path='models/granite-7b-lab-Q4_K_M.gguf'
```

this results into creating the following `config.yaml`:

```yaml
chat:
  context: default
  greedy_mode: false
  logs_dir: data/chatlogs
  max_tokens: null
  model: models/granite-7b-lab-Q4_K_M.gguf
  session: null
  vi_mode: false
  visible_overflow: true
general:
  log_level: INFO
generate:
  chunk_word_count: 1000
  model: models/granite-7b-lab-Q4_K_M.gguf
  num_cpus: 10
  num_instructions: 100
  output_dir: generated
  prompt_file: prompt.txt
  seed_file: seed_tasks.json
  taxonomy_base: origin/main
  taxonomy_path: taxonomy
serve:
  gpu_layers: -1
  host_port: 127.0.0.1:8000
  max_ctx_size: 4096
  model_path: models/granite-7b-lab-Q4_K_M.gguf
```

2. Proceed with the download phase:

```bash
ilab download

Downloading model from instructlab/merlinite-7b-lab-GGUF@main to models...
Downloading 'merlinite-7b-lab-Q4_K_M.gguf' to 'models/.huggingface/download/merlinite-7b-lab-Q4_K_M.gguf.9ca044d727db34750e1aeb04e3b18c3cf4a8c064a9ac96cf00448c506631d16c.incomplete'
INFO 2024-06-14 04:04:10,713 file_download.py:1877: _download_to_tmp_and_move Downloading 'merlinite-7b-lab-Q4_K_M.gguf' to 'models/.huggingface/download/merlinite-7b-lab-Q4_K_M.gguf.9ca044d727db34750e1aeb04e3b18c3cf4a8c064a9ac96cf00448c506631d16c.incomplete'
merlinite-7b-lab-Q4_K_M.gguf: 100%|███████████████████████████████████████████████████████████████████████████████████████████| 4.37G/4.37G [00:56<00:00, 76.7MB/s]
Download complete. Moving file to models/merlinite-7b-lab-Q4_K_M.gguf

INFO 2024-06-14 04:05:07,967 file_download.py:1893: _download_to_tmp_and_move Download complete. Moving file to models/merlinite-7b-lab-Q4_K_M.gguf
```

As you can see it's downloading `merline` instead of `granite` (which I have initialized my devenv). My expectation is that it should have downloaded granite instead, even without me having to explicitely specify it (e.g. via `ilab download --repository instructlab/granite-7b-lab-GGUF --filename granite-7b-lab-Q4_K_M.gguf --model-dir models` as this PR suggests). I have this expectation because in my `config.yaml` I have absolutely nothing about `merline` and the initialization phase has already passed, no there is no excuse saying "this is the default" because I have bypassed the default.

Am I missing something? Might be, this is my first try actually of InstructLab. The only thing I am thinking which I might be wrong, is that during the `init --help` it says:

```
--model-path PATH               Path to the model used during generation.
```

So this is used only for the generation? If so the name `--model-path` should be renamed into reflecting this (e.g. `--model-gen-path`). Because if that's the case, it give false impression.

At the same time, I am not sure what the `download` command is supposed to do, because the manual says:

```
Download the model(s) to train
```

So it downloads a model, which is not related to the `init` phase, and neither with the `config.yaml`? Shouldn't it download all the models listed in the `config.yaml` for that matter?

Sorry for the wall of text. Since you are going through this docs, I thought to give you my perspective on that :)

File: README.md [hunk 4]
```diff
@@ -172,7 +172,7 @@ For an overview of the full workflow, see the [workflow diagram](./docs/workflow
       python3 -m venv --upgrade-deps venv
       source venv/bin/activate
       (venv) $ pip cache remove llama_cpp_python
-      (venv) $ pip install instructlab -C cmake.args="-DLLAMA_CUDA=on"
+      (venv) $ pip install instructlab -C cmake.args="-DLLAMA_CUBLAS=on"
```
Comment (tiran) [inline README.md:175]: Please undo this change. `LLAMA_CUDA` is the correct option. `LLAMA_CUBLAS` is deprecated.
```suggestion
      (venv) $ pip install instructlab -C cmake.args="-DLLAMA_CUDA=on"
```

**Review by tiran — CHANGES_REQUESTED** [review]
